systems/subscriber/sim-manager/pkg/utils/prometheus_utils.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/sirupsen/logrus"
+)
+
+func PushMetrics(metricJob, metricName string, metricHelp string, metricType prometheus.ValueType, metricLabels map[string]string, metricValue float64) {
+	labelDimensions := make([]string, 0, len(metricLabels))
+
+	for key := range metricLabels {
+		labelDimensions = append(labelDimensions, key)
+	}
+
+	var metric prometheus.Collector
+
+	switch metricType {
```

Comment:
Here you would be creating metric collector for each push which is not right. May be try to maintain list of metric collectors which you have created. So before pushing check if metric collector exist. If it exisst get the metric collector and set the value otherwise create a new metric collector.